inlanecruising_plugin/include/inlanecruising_plugin/smoothing/SplineI.h

```diff
@@ -41,15 +41,15 @@ class SplineI
    * \param points The key points
    */ 
   virtual void setPoints(std::vector<lanelet::BasicPoint2d> points) = 0;
-
-  /**
-   * \brief Get the y value for the given x value
+  
+    /**
+   * \brief Get the Eigen::VectorXf for the given t parameter
    * 
-   * \param x The value to solve the spline for
+   * \param t The value to solve the spline for
    * 
-   * \return The y value that matches x
+   * \return The Eigen::VectorXf vector with x, y that matches t parameter
    */ 
-  virtual double operator()(double x) const = 0;
+  virtual Eigen::VectorXf operator()(double t) const = 0;
```

Comment:
If the intent is to use this class as an abstraction layer over top of the spline implementation, does it make sense to use an Eigen-specific type here? I'm kind of split personally, Eigen is the underlying implementation we're abstracting over, but also just a support library we'd be using regardless.

Reply:
I'll make it std::pair